### PR TITLE
docs(troubleshooting): add decorator troubleshooting guide

### PR DESCRIPTION
**Pull request by joielee09**
Add the example and solution for the decorator-legacy issue.

### CODE REVIEW

**Comment by nicojs** [comment]
Thanks! 🎉 

I've changed the wording a bit and added an explanation of what is wrong.

**Comment by joielee09** [comment]
@nicojs Thank you for your feedback. I definitely see the difference. I only describe 'How'. On the other hand, you explained 'Why'. Why it happens and why it is fixed with an alternative workaround. I feel a little embarrassing for how little did I know. When I meet issues I will find the answer in the way you do it. : )

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -172,37 +172,38 @@ FATAL ERROR: Committing semi space failed. Allocation failed - JavaScript heap o
...
-In order to solve this, you need to make sure your project have babel and the decorator-legacy plugin setting. For example,
-```
-    plugins: [
-      ["@babel/plugin-proposal-decorators", { legacy: true }],
-      ["@babel/plugin-proposal-class-properties", { loose: true }],
-    ],
-```
-See [here](https://babeljs.io/docs/en/babel-plugin-proposal-decorators) for babel decorators.
-You also have to let Stryker notice the plugins. Put this code in your stryker.conf.json.
+Stryker JS uses [babel](https://babeljs.io/docs/en/) to parse and mutate your files. When parsing your your `.js` files, it reads in the `.babelrc` and `babel.config.js` files you have, but applies its own default plugins first. In rare circumstances, those default plugins are incompatible with your babel configuration.
```
Comment (joielee09) [inline docs/troubleshooting.md:195]: There's 'your' written twice in ' When parsing your your `.js` files,' : )

**Comment by nicojs** [comment]
Haha no problem. I'm one of the maintainers, so it makes sense that I can explain why. I'm very grateful for the work you did 😊

**Pull request merged**